llvm/lib/Target/Xtensa/XtensaISelLowering.cpp

```diff
@@ -827,9 +1010,70 @@ XtensaTargetLowering::emitSelectCC(MachineInstr &MI,
 
 MachineBasicBlock *XtensaTargetLowering::EmitInstrWithCustomInserter(
     MachineInstr &MI, MachineBasicBlock *MBB) const {
+  const TargetInstrInfo &TII = *Subtarget.getInstrInfo();
+  DebugLoc DL = MI.getDebugLoc();
+
   switch (MI.getOpcode()) {
   case Xtensa::SELECT:
     return emitSelectCC(MI, MBB);
+  case Xtensa::SHL_P: {
```

Comment:
Instruction selection should be done in `XtensaDAGToDAGISel::Select`.

Reply:
I added "usesCustomInserter" attribute to the  SHL_P and some other shift pseudo operations. Should I remove this operations processing from EmitInstrWithCustomInserter or maybe remove "usesCustomInserter" attribute?

Reply:
This code should be moved to XtensaISelDAGToDAG.cpp and SHL_P and others removed.

Reply:
fixed